CAS-366 - Populate PP details from nDelius

froddd opened the pull request:
# Context

https://dsdmoj.atlassian.net/browse/CAS-366

# Changes in this PR

The screen used to collect the Probation Practitioner's details now shows the following user details as they are in nDelius:

- Name
- Email
- Phone number
- PDU

Each detail is accompanied by a 'Change' link, which lets the user update this information for the current application, and offers a link to request an update of their nDelius details in an external service if incorrect. If there is no detail available from nDelius, a link is shown to simply enter the missing information, using the same screen as to update it.

All details are required, so the 'Save and continue' button is only shown once all have been provided.

The screen to add/update the PDU uses the [accessible autocomplete from AlphaGov](https://github.com/alphagov/accessible-autocomplete?tab=readme-ov-file#progressive-enhancement) to progressively enhance the `<select>` element.

## Screenshots of UI changes

### Before

#### PP details:

<img width="796" alt="Screenshot 2024-07-24 at 13 35 10" src="https://github.com/user-attachments/assets/0f67ea2d-bb35-4986-b69a-9b468a4cf13a">

### After

#### Confirm PP details:
<img width="803" alt="Screenshot 2024-07-24 at 13 33 19" src="https://github.com/user-attachments/assets/533893aa-974c-454e-ba97-2658008abf8e">

#### Update PP phone number (similar screen for Name and Email):

<img width="702" alt="Screenshot 2024-07-25 at 12 40 57" src="https://github.com/user-attachments/assets/44127f30-780b-4039-ba6e-aef09c221df7">

#### Update PP PDU:

<img width="791" alt="Screenshot 2024-07-25 at 12 41 25" src="https://github.com/user-attachments/assets/1e73f4fc-7455-4df3-9599-f8ddb6da9206">



# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

<!-- Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible. You can monitor deployment failures in CircleCI
itself and application errors are found in
[Sentry](https://ministryofjustice.sentry.io/issues/?project=4504129156218880&referrer=sidebar&statsPeriod=24h).
Both events should be automatically sent to our [Slack
channel](https://mojdt.slack.com/archives/C048BJS7S2F). -->

Commit:
Implement PP detail change form pages

These new screens save data alongside the main application contact
details data, but do not produce any output for the final document. The
main Probation Practitioner details screen uses this data to display the
most up-to-date information.

Commit:
Update main PP details screen

Details are now shown as a list, with a set of links to add/update any
information using the newly created forms.

Commit:
Update PDU mock data to include realistic ids

Commit:
Update user service to include extra properties

This adds email, telephoneNumber and probationDeliveryUnit to the
userDetails object.

Commit:
Only show Save button if all details provided

aliuk2012 commented on server/views/applications/pages/accommodation-need/contact-details/practitioner-email.njk:
```diff
@@ -0,0 +1,22 @@
+        label: {
+            text: "Email address"
+        }
+    }, fetchContext()) }}
```
Could we start labeling input fields using the page heading instead of "Email address"? Without the page heading theres no context what the user is meant to record. In this case, "Email address"  adds very little value and it would better to use the page heading "Update probation practitioner email address".

aliuk2012 replied:
I've added a comment on the ticket to ask Ray for his opinion on this.

aliuk2012 commented on server/views/applications/pages/accommodation-need/contact-details/practitioner-name.njk:
```diff
@@ -0,0 +1,22 @@
+    {{ formPageInput({
+        fieldName: "name",
+        label: {
+            text: "Full name"
```
Same comment as above

aliuk2012 commented on server/views/applications/pages/accommodation-need/contact-details/practitioner-phone.njk:
```diff
@@ -0,0 +1,22 @@
+    {{ formPageInput({
+        fieldName: "phone",
+        label: {
+            text: "Phone number"
```
Same comment as above

Commit:
Pre-populate PDU with user details

PDU is now a select with the PDU options provided from the API. Selected
option defaults to the user's PDU if it exists in the session.

Commit:
Move PDU info to main PP details screen

PDU now follows the same pattern as Name, Email and Phone: data comes
from user details by default, but can be updated using a separate
screen. The PDU now shows alongside those other details in the
application document.

Commit:
Use an accessible autocomplete for PDU select

A hint is inserted after the label to avoid using a less accessible
placeholder.

aliuk2012 commented on server/views/applications/pages/accommodation-need/contact-details/probation-practitioner.njk:
```diff
@@ -1,12 +1,24 @@
+        text: 'You need to update anything that is incorrect or missing.'
+    }) }}
 
-  {% include "../../partials/_contact-details.njk" %}
```
I was wondering, if this partial was still needed so searched the code base to see its only mentioned once here https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/blob/main/server/views/applications/pages/accommodation-need/contact-details/backup-contact.njk#L10. If its only being used once then could we get rid of the partial and move its contents to the one place it is referenced.

froddd replied:
Oooh that's a good spot!

Commit:
Content changes for update screens

Use main heading as label, removing the need for a separate label, for
better accessibility. Phone number field has been resized to better
indicate expected content.

aliuk2012 reviewed (approved):
👍  Nice work
